File: browserfleet/server/sessions/enums.py
import enum

class OperatingSystem(enum.Enum):
    """Operating systems"""
    WINDOWS = "windows"
    MACOS = "macos"
    LINUX = "linux"

class Browser(enum.Enum):
    """Browser types"""
    CHROME = "chrome"
    FIREFOX = "firefox"
    EDGE = "edge"
    SAFARI = "safari"

class BrowserVersion(enum.Enum):
    """Browser versions"""
    LATEST = "latest"
    STABLE = "stable"
    CANARY = "canary"
    DEV = "dev"

class SessionStatus(enum.Enum):
    """Status for a browser session"""
    PENDING = "pending"
    STARTING = "starting"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"
    EXPIRED = "expired"
    CRASHED = "crashed"
    TIMED_OUT = "timed_out"
    TERMINATED = "terminated"

class SessionEventType(enum.Enum):
    """Event types for browser session lifecycle"""
    # Session initialization events
    SESSION_CREATED = "session_created"       # Session record created in database
    SESSION_ASSIGNED = "session_assigned"     # Session assigned to a worker
    SESSION_STARTING = "session_starting"     # Browser process is starting
    BROWSER_STARTED = "browser_started"       # Browser successfully started
    
    # Session runtime events
    SESSION_IDLE = "session_idle"             # No activity detected for some time
    SESSION_ACTIVE = "session_active"         # Activity resumed after idle
    
    # Session termination events
    SESSION_COMPLETED = "session_completed"   # Session ended normally
    SESSION_CRASHED = "session_crashed"       # Browser crashed
    SESSION_TIMED_OUT = "session_timed_out"   # Session exceeded time limit
    SESSION_TERMINATED = "session_terminated" # Session forcibly terminated

File: browsergrid/internal/sessions/enums.go
package sessions

// OperatingSystem represents supported operating systems
// @Description Supported operating systems for browser sessions
type OperatingSystem string //@name OperatingSystem

const (
	OSWindows OperatingSystem = "windows"
	OSMacOS   OperatingSystem = "macos"
	OSLinux   OperatingSystem = "linux"
)

// Browser represents supported browser types
// @Description Supported browser types
type Browser string //@name Browser

const (
	BrowserChrome   Browser = "chrome"
	BrowserChromium Browser = "chromium"
	BrowserFirefox  Browser = "firefox"
	BrowserEdge     Browser = "edge"
	BrowserOpera    Browser = "webkit"
	BrowserSafari   Browser = "safari"
)

// BrowserVersion represents browser version types
// @Description Browser version types (latest, stable, canary, dev)
type BrowserVersion string //@name BrowserVersion

const (
	VerLatest BrowserVersion = "latest"
	VerStable BrowserVersion = "stable"
	VerCanary BrowserVersion = "canary"
	VerDev    BrowserVersion = "dev"
)

// SessionStatus represents the current status of a browser session
// @Description Current status of a browser session
type SessionStatus string //@name SessionStatus

const (
	StatusPending    SessionStatus = "pending"
	StatusStarting   SessionStatus = "starting"
	StatusAvailable  SessionStatus = "available"
	StatusClaimed    SessionStatus = "claimed"
	StatusRunning    SessionStatus = "running"
	StatusIdle       SessionStatus = "idle"
	StatusCompleted  SessionStatus = "completed"
	StatusFailed     SessionStatus = "failed"
	StatusExpired    SessionStatus = "expired"
	StatusCrashed    SessionStatus = "crashed"
	StatusTimedOut   SessionStatus = "timed_out"
	StatusTerminated SessionStatus = "terminated"
)

// SessionEventType represents different types of session events
// @Description Types of events that can occur during a session
type SessionEventType string //@name SessionEventType

const (
	EvtSessionCreated    SessionEventType = "session_created"
	EvtResourceAllocated SessionEventType = "resource_allocated"

	EvtSessionStarting  SessionEventType = "session_starting"
	EvtContainerStarted SessionEventType = "container_started"
	EvtBrowserStarted   SessionEventType = "browser_started"
	EvtSessionAvailable SessionEventType = "session_available"

	EvtSessionClaimed  SessionEventType = "session_claimed"
	EvtSessionAssigned SessionEventType = "session_assigned"
	EvtSessionReady    SessionEventType = "session_ready"

	EvtSessionActive SessionEventType = "session_active"
	EvtSessionIdle   SessionEventType = "session_idle"
	EvtHeartbeat     SessionEventType = "heartbeat"

	EvtPoolAdded   SessionEventType = "pool_added"
	EvtPoolRemoved SessionEventType = "pool_removed"
	EvtPoolDrained SessionEventType = "pool_drained"

	EvtSessionCompleted  SessionEventType = "session_completed"
	EvtSessionExpired    SessionEventType = "session_expired"
	EvtSessionTimedOut   SessionEventType = "session_timed_out"
	EvtSessionTerminated SessionEventType = "session_terminated"

	EvtStartupFailed     SessionEventType = "startup_failed"
	EvtBrowserCrashed    SessionEventType = "browser_crashed"
	EvtContainerCrashed  SessionEventType = "container_crashed"
	EvtResourceExhausted SessionEventType = "resource_exhausted"
	EvtNetworkError      SessionEventType = "network_error"

	EvtStatusChanged SessionEventType = "status_changed"
	EvtConfigUpdated SessionEventType = "config_updated"
	EvtHealthCheck   SessionEventType = "health_check"
)
